src/main/integrations/slack/index.ts

```diff
@@ -0,0 +1,78 @@
+import playerStateStore, { PlayerState, VideoDetails, VideoState } from "../../player-state-store";
+import IIntegration from "../integration";
+import { StoreSchema } from "~shared/store/schema";
+import Conf from "conf";
+import { PostType } from "../../@types/slack";
+
+export default class SlackStatus implements IIntegration {
+  private store: Conf<StoreSchema>;
+  private enabled = false;
+  private stateCallback: (event: PlayerState) => void = null;
+  private currentVideoId: string = "";
+  private statusTimeoutId: NodeJS.Timeout;
+  private slackUserToken: string = "";
+
+  private async playerStateChanged(state: PlayerState) {
+    if (!this.enabled) return;
+    const { secondsTillClearStatus } = await this.store.get("slack");
+    if (state.trackState === VideoState.Playing && state.videoDetails) {
+      if (!state.adPlaying) {
```

Comment:
Any reason to not combine these 2 statements or do some early exits instead.
Feels like it would clean up from having a spaghetti of If statements a bit more. ❤

```js
if (!this.enabled ||
    !state.adPlaying ||
    !state.videoDetails ||
    state.trackState !== VideoState.Playing
 ) return;

Reply:
i like what you sent, but i did something similar